core/templates/pages/exploration-player-page/layout-directives/content-language-selector.component.ts

```diff
@@ -61,14 +63,22 @@ export class ContentLanguageSelectorComponent implements OnInit {
   currentGlobalLanguageCode!: string;
 
   ngOnInit(): void {
+    const url = new URL(this.windowRef.nativeWindow.location.href);
     this.currentGlobalLanguageCode = (
       this.i18nLanguageCodeService.getCurrentI18nLanguageCode());
     this.selectedLanguageCode = (
       this.contentTranslationLanguageService.getCurrentContentLanguageCode());
     this.languageOptions = (
       this.contentTranslationLanguageService.getLanguageOptionsForDropdown());
     for (let option of this.languageOptions) {
-      if (option.value === this.currentGlobalLanguageCode) {
+      if (url.searchParams.get('initialContentLanguageCode') === option.value) {
```

Comment:
I am a bit confused about the logic here, it seems like the if and else blocks execute the same code, is that intentional?

Reply:
@kevintab95 I have updated the logic so that we can avoid code duplication. PTAL and let me know for further changes.